File: mergeSort/mergeSort.cpp
#include <iostream>
using namespace std;

void Merge(int *arr, int form, int to, int mid)
{
	int i = form;
	int j = mid+1;
	int k = 0;

	int *temp = new int[to-form+1];

	while(i <= mid && j<=to)
	{
		if(arr[i]<arr[j])
		{
			temp[k] = arr[i];
			k++;i++;
			
		}
		else
		{
			temp[k] = arr[j];
			k++;j++;
			
		}
	}

	while(i<=mid)
	{
		temp[k] = arr[i];
		k++;i++;
		
	}
	while(j<=to)
	{
		temp[k] = arr[j];
		k++;j++;
		
	}
	for(int i=form; i <= to;i++)
	{
		arr[i] = temp[i -form];
	}
}

void MergeSort(int *arr, int form, int to)
{
	int mid;
	if(form < to)
	{
		mid = (form+to)/2;
		MergeSort(arr, form, mid);
		MergeSort(arr, mid+1, to);

		Merge(arr, form, to, mid);
	}
}

int main()
{
	cout<<"需要输入的数据个数： ";
	int n;
	cin >> n;
	int *arr = new int[n];
	cout <<"输入需要排序的数据： "<<endl;
	for(int i=0;i < n; i++)
	{
		cin >> arr[i];
	}

	MergeSort(arr, 0,n-1);
	
	for(int i=0; i < n; i++)
		cout<<arr[i]<<" ";

	return 0;
}

File: mergeSort/mergeSort.py
def merge(left,right):
    if not len(left) or not len(right):
        return left or right
    
    result = []
    i,j=0,0
    
    while(len(result) < len(left)+len(right)):
        if left[i] < right[j]:
            result.append(left[i])
            i+=1
        else:
            result.append(right[j])
            j+=1
        if i == len(left) or j == len(right):
            result.extend(left[i:] or right[j:])
            break
        
    return result



def merge_Sort(nlist):
    if len(nlist) < 2:
        return nlist
    
    mid = int(len(nlist)/2)
    left = merge_Sort(nlist[:mid])
    right = merge_Sort(nlist[mid:])
    
    return merge(left, right)

nnlist = input("请输入需要排序的数组： ")
nlist = [int(n) for n in nnlist.split()]
print(merge_Sort(nlist))
 
